store/displayPreferences.ts

```diff
@@ -1,116 +1,108 @@
-import Vue from 'vue';
 import { DisplayPreferencesDto } from '@jellyfin/client-axios';
-import { MutationTree, ActionTree, GetterTree } from 'vuex';
+import { MutationTree, ActionTree } from 'vuex';
+import { merge, toString as toStr } from 'lodash';
+import { boolean } from 'boolean';
 import nuxtConfig from '~/nuxt.config';
 
-const stringToBoolean = (value: string): boolean => value === 'True';
-const booleanToString = (value: boolean): string => (value ? 'True' : 'False');
-
-export interface DisplayPreferencesState extends DisplayPreferencesDto {
-  CustomPrefs: {
-    [key: string]: string;
-  };
+/**
+ * TODO: Generate this from the nuxtConfig file
+ */
+export enum SupportedLocales {
+  auto = 'auto',
+  cs = 'cs',
+  de = 'de',
+  enUs = 'enUs',
+  es = 'es',
+  fr = 'fr',
+  nb = 'nb',
+  nl = 'nl',
+  pl = 'pl',
+  ro = 'ro',
+  sk = 'sk',
+  sl = 'sl',
+  sv = 'sv',
+  ta = 'ta',
+  tr = 'tr',
+  vi = 'vi',
+  zhCn = 'zhCn'
 }
```

Comment:
Unless we find a way to generate it, I find it superfluous as it's only used to type a little bit more and increases the chance of missing a new locale :/

Reply:
We can request it from the server, I havent looked at this PR much so I don't know if that's useful or not

Reply:
Server uses different locale codes as we do, as we need to comply with the datefns schema to ease the translation of dates.

Reply:
Also we don't necessarily use the same locales as the server, do we?

Reply:
Exactly

Reply:
Personally I'd just drop the enum, wdyt?